tests/sweep_framework/sweeps/all_gather_T3K_sharded.py

```diff
@@ -0,0 +1,239 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+
+import torch
+
+import ttnn
+from models.utility_functions import get_devices_for_t3000
+from tests.ttnn.utils_for_testing import check_with_pcc, start_measuring_time, stop_measuring_time
+from loguru import logger
+import tt_lib as ttl
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from tests.ttnn.unit_tests.operations.test_all_gather import is_unsupported_case
+from ttnn import ShardTensorToMesh
+
+# Override the default timeout in seconds for hang detection.
+TIMEOUT = 30
+
+# Parameters provided to the test vector generator are defined here.
+# They are defined as dict-type suites that contain the arguments to the run function as keys, and lists of possible inputs as values.
+# Each suite has a key name (in this case "suite_1" and "suite_2") which will associate the test vectors to this specific suite of inputs.
+# Developers can create their own generator functions and pass them to the parameters as inputs.
+parameters = {
+    "all_gather": {
+        "num_devices": [4, 8],
+        "num_links": [1, 2],
+        "input_shape": [
+            # [1, 1, 32, 32], # throwing python segmentation fault
+            [1, 1, 32, 1024],
```

Comment:
We need much more comprehensive coverage here. If it's easier as separate parameter dicts, that's fine, but we need to cover a lot more cases.

From the master issue:
> - Shapes: Constrained to tile/page aligned, Shard grids unpadded
  - For inner dims (y, x), increment by 32 in each direction. For outer dims, increment by 1.
    - Outer dims can be swept by basic numbers, then relatively prime numbers up to 128.
More values can be swept over after all-gather is migrated to make more use of runtime args.